modules/ding_mobilesearch/ding_mobilesearch.module

```diff
@@ -117,61 +117,174 @@ function _ding_mobilesearch_types() {
+  $paragraphs_field = NULL;
+  // Content type specific fields.
+  switch ($node->type) {
+    case 'ding_event':
+      $items = field_get_items('node', $node, 'field_ding_event_date');
+      if (is_array($items) && !empty($items)) {
+        $item = reset($items);
+
+        $field_field = field_info_field('field_ding_event_date');
+        $field_instance = field_info_instance('node', 'field_ding_event_date', 'ding_event');
+
+        $from_object = new DateObject($item['value'], 'UTC', date_type_format($field_field['type']));
+        $from_object->setTimezone(new DateTimeZone($tz));
+        $to_object = new DateObject($item['value2'], 'UTC', date_type_format($field_field['type']));
+        $to_object->setTimezone(new DateTimeZone($tz));
+
+        $all_day = date_all_day_field($field_field, $field_instance, $from_object, $to_object);
+
+        $mapping['fields']['field_ding_event_date'] = array(
+          'name' => 'Event date',
+          'value' => array(
+            'from' => ding_mobilesearch_format_date($from_object->getTimestamp(), $item['timezone']),
+            'to' => ding_mobilesearch_format_date($to_object->getTimestamp(), $item['timezone']),
+          ),
+          'attr' => array(
+            'all_day' => $all_day,
+          ),
+        );
+      }
+      $tag_field = 'field_ding_event_tags';
+      break;
+
+    case 'ding_news':
+      $tag_field = 'field_ding_news_tags';
+      $paragraphs_field = 'field_ding_news_paragraphs';
+      break;
+
+    case 'ding_page':
+      $tag_field = 'field_ding_page_tags';
+      $paragraphs_field = 'field_ding_page_paragraphs';
+      break;
+  }
+
+  if ($tag_field) {
+    $mapping['fields'][$tag_field] = array(
+      'name' => 'Tags',
+      'value' => array(),
+      'attr' => array(),
+    );
+
+    $items = field_get_items('node', $node, $tag_field);
+    if (is_array($items) && !empty($items)) {
+      foreach ($items as $item) {
+        if ($term = taxonomy_term_load($item['tid'])) {
+          $mapping['fields'][$tag_field]['value'][] = $term->name;
+        }
+      }
+    }
+  }
+
+  // Paragraphs.
+  if ($paragraphs_field) {
+    $mapping['fields'][$paragraphs_field] = [
+      'name' => 'Paragraphs',
+      'value' => [],
+      'attr' => [],
+    ];
+    $node_wrapper = entity_metadata_wrapper('node', $node);
+    $paragraphs_item_entities = $node_wrapper->{$paragraphs_field}->value();
+
+    $mapping['fields'][$paragraphs_field]['value'] = array_map(function ($paragraphs_item_entity) {
+      $bundle = $paragraphs_item_entity->bundle();
+      $mapping = _ding_mobilesearch_map_entity_fields('paragraphs_item', $bundle, $paragraphs_item_entity);
+      return [
+        'type' => $bundle,
+        'fields' => $mapping['fields'],
+      ];
+    }, $paragraphs_item_entities);
+  }
+
+  return $mapping;
+}
+
+/**
+ * Helper function to map common fields from an entity.
+ */
+function _ding_mobilesearch_map_entity_fields($entity_type, $bundle, $entity) {
```

Comment:
:+1: but remember to document function arguments.

Reply:
Added documentation.